python/cudf/dataframe/multiindex.py

```diff
@@ -199,7 +202,22 @@ def _compute_validity_mask(self, df, row_tuple):
         return validity_mask
 
     def _get_row_major(self, df, row_tuple):
-        valid_indices = self._compute_validity_mask(df, row_tuple)
+        print(row_tuple)
+        slice_access = False
+        if isinstance(row_tuple[0], numbers.Number):
+            valid_indices = row_tuple[0]
+        elif isinstance(row_tuple[0], slice):
+            # 1. empty slice compute
+            if row_tuple[0].stop == 0:
+                valid_indices = []
+            else:
+                slice_access = True
+                start = row_tuple[0].start or 0
+                stop = row_tuple[0].stop or len(df)
+                step = row_tuple[0].step or 1
+                valid_indices = np.array(range(start, stop, step))
```

Comment:
Instead of using a numpy array here you should use the `cudautils.arange(...)`

Reply:
I worried about that line specifically.